mlflow/transformers.py

```diff
@@ -1571,7 +1571,50 @@ def _convert_pandas_to_dict(self, data):
                     )
             return parsed
 
-    def predict(self, data, device=None):
+    def _override_inference_config(self, **kwargs):
+        if kwargs:
+            _logger.warning(
+                "kwargs provided to the `predict` method will override the inference configuration "
+                "saved with the model. If the kwargs provided are not valid for the pipeline, "
+                "MlflowException will be raised."
+            )
+
+        # Override the inference configuration with any additional kwargs provided by the user.
+        for key, value in kwargs.items():
+            self.inference_config[key] = value
+
+    def _validate_inference_config_and_return_output(self, data):
+        import transformers
+
+        try:
+            if isinstance(data, dict):
+                return self.pipeline(**data, **self.inference_config)
+            return self.pipeline(data, **self.inference_config)
+        except ValueError as e:
+            if "The following `model_kwargs` are not used by the model" in str(e):
+                raise MlflowException(
+                    "The kwargs provided to the `predict` method are not valid "
+                    f"for pipeline {type(self.pipeline)}. Caused by: {repr(e)}",
```

Comment:
```suggestion
                    f"for pipeline {type(self.pipeline).__name__}. Caused by: {repr(e)}",
```
I think having just the name of the Pipeline is a bit more user-friendly as the namespace pathing in some of the implementations can be quite long and distracting